### ISSUE TITLE
Feature : Assets groups with different options ?

### ISSUE THREAD

**Issue by wmcmurray** [JeffreyWay/laravel-mix#1359]
Hi (this is a question not an issue)

Could it be possible to implement a way to create groups of assets on which we could define specific options instead of only having access to global options ? :thinking: https://github.com/JeffreyWay/laravel-mix/blob/master/src/Api.js#L483

### My concrete use case is : 
I work for a company which has a website and a custom backend/cms. We just added a *super-cool-new-user-interface* using VueJS. Now we want to send ourselves slack notifications when a client experiences JavaScript errors.

Therefore we need to generate sourcemaps in production (to have useful stacktraces), but there is no easy way to generate sourcemaps ONLY for the `.js` files used in this new user-interface. We are forced to generate sitemaps for ALL assets of ALL parts of our applications and this takes up quite some times ! :/

I managed to hack something that works, but I was wondering if it could be done directly in laravel-mix ? 

Thanks ! :slightly_smiling_face:

**Comment by FaridAghili** [comment]
AFAIK it's not possible.

Can I ask you to explain how "you managed to hack something that works"?
Need it for a similar case.

**Comment by wmcmurray** [comment]
(sorry for the delay, I was in vacation :palm_tree: )

@FaridAghili it's too disparate to be copy/pasted, so I'll try to explain it...

First, I must say that we already had altered the webpack.mix.js to counter the problem when we try to build assets in production while clients are accessing the website (it caused mix file not found errors because the manifest file reference files that have not been re-builded yet). To counter that we had to implement more steps while building assets : first we copy the currently builded assets directory into a new one, we re-write the current manifest file to point to that new directory, then we change the filename of the new manifest we are about to build ( `Mix.manifest.name = 'mix-manifest-new.json';` ). After it's finished, we move the new manifest in place of the old one and delete the old assets dir.

Based on that, I separated each assets into groups (ex 'frontend', 'backend', etc) which I can build all at the same time using `npm run prod`, or separately using environment variables in front of the command. This way I can enable the generation of sourcemaps when we are specifically building only the group in which we want sourcemaps generated. Now it gets complicated because the manifest file generated will contain only the assets inside the specific group we are building ! So I had to merge the new manifest into the old one to keep track of builded assets that are considered "unchanged". (but it was not so hard because we already had a system that separated the old manifest from the new one).

In the end, this makes the webpack.mix file very complicated to understand and everytime I build assets in production I'm afraid that everything will explode... but so far it has not happened ! :crossed_fingers: 

Hope it helps !
